### DISCUSSION TITLE
Add ImageNode

### DISCUSSION THREAD

**Discussion by Pabs14324** [CariusLars/ar_flutter_plugin#42] [Ideas]
Hi. I've been working on a school project that would allow users to put 2d and 3d assets via cloud on the AR scene. Been working on this for almost a year now and your plugin made the 3d part work. Now I am stuck with the 2d part. I was able to make the 2d part work using the arcore_flutter_plugin and path_provider. I don't how you make these plugins but I hope we would be able to add 2d assets on the AR scene similar to arcore_flutter_plugin's 'ArCoreImage()' capability.

**Comment by CariusLars** [comment]
Hey @Pabs14324 , thanks for your message! 
It would be great if you contributed that functionality to the AR Flutter Plugin, so here's some help on how to do this for the Android side:

I would suggest adding a new [NodeType](https://github.com/CariusLars/ar_flutter_plugin/blob/main/lib/datatypes/node_types.dart), e.g. `2dImage`, then create a subclass of [ARNode](https://github.com/CariusLars/ar_flutter_plugin/blob/main/lib/models/ar_node.dart), something like an `ARImageNode` with fields like width, height, and the actual image data in bytes.

To process the new type of node on the platform side, add a node type case to the `addNode` function of the [AndroidARView](https://github.com/CariusLars/ar_flutter_plugin/blob/main/android/src/main/kotlin/io/carius/lars/ar_flutter_plugin/AndroidARView.kt). To process the new node type, you should then add a new 2d image node builder function to the [ArModelBuilder](https://github.com/CariusLars/ar_flutter_plugin/blob/main/android/src/main/kotlin/io/carius/lars/ar_flutter_plugin/ArModelBuilder.kt), you can probably just copy most of the code from [ARCoreFluterPlugin's Renderable Factory](https://github.com/giandifra/arcore_flutter_plugin/blob/master/android/src/main/kotlin/com/difrancescogianmarco/arcore_flutter_plugin/RenderableCustomFactory.kt) and make it return a node which can then be processed just like any other node in the plugin.

The iOS side of things will look very similar, but it's probably good to start with one OS first and then add the other.

I hope this gets you pointed into the right direction, if you branch or fork this repo and work on the new feature, I'm happy to provide support and feedback!

Cheers

Lars

**Comment by 5hanth** [comment]
did anyone manage to render a 2D image in ARView.
